pkg/rpc/server/server.go

```diff
@@ -42,7 +50,7 @@ func (s *ScanServer) Scan(ctx context.Context, in *rpcScanner.ScanRequest) (*rpc
 	}
 	results, os, err := s.localScanner.Scan(ctx, in.Target, in.ArtifactId, in.BlobIds, options)
 	if err != nil {
-		return nil, xerrors.Errorf("failed scan, %s: %w", in.Target, err)
+		return nil, WarnAndError("failed scan, %s: %w", in.Target, err)
```

Comment:
I'd wrap errors in the caller as the main reason for using `xerrors` is the stack trace. If we wrap errors in `WanAndError`, the stack trace shows the line number in `WranAndError`.
https://go.dev/play/p/-QPSArNPmX6

```suggestion
		return nil, WarnAndError(xerrors.Errorf("failed scan, %s: %w", in.Target, err)
```